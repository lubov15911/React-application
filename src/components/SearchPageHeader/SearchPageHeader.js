import React from 'react';
import PropTypes from 'prop-types';

import HomePageLogoButton from '../HomePageLogoButton/HomePageLogoButton';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import SearchHeader from '../SearchHeader/SearchHeader';

import './SearchPageHeader.scss';

const SearchPageHeader = (props) => {
    const {
        routeToHomePage,
    } = props;

    return (
        <div className="header-container">
            <HomePageLogoButton handleClick={routeToHomePage} />
            <ErrorBoundary>
                <SearchHeader  />
            </ErrorBoundary>
        </div>
    );
};
SearchPageHeader.propTypes = {
    routeToHomePage: PropTypes.func.isRequired,
};

export default SearchPageHeader;
