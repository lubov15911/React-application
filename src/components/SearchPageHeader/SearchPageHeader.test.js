import React from 'react';
import { shallow } from 'enzyme';

import SearchPageHeader from './SearchPageHeader';

describe('SearchPageHeader', () => {
    it('should render correctly', () => {
        const fakeFunction = () => {};

        const component = shallow(<SearchPageHeader routeToHomePage={fakeFunction} />);
        expect(component).toMatchSnapshot();
    });
});
