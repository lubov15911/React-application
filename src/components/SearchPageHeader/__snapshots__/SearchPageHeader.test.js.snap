// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPageHeader should render correctly with props 1`] = `
<div
  className="header-container"
>
  <HomePageLogoButton
    handleClick={[Function]}
  />
  <ErrorBoundary>
    <SearchHeader />
  </ErrorBoundary>
</div>
`;
